Added support for NSAttributedString.

Remki opened the pull request:
Hello

I added support for attributed strings + refactored code a bit, to react on font changed (in all views except button, where i only support attributed strings).
I also updated project settings - example application was not building.

I hope you like it, but if you have any comments or ideas  how to do this better i will fix it as you wish.

Grubas7 commented:
Would be great to see some tests.

Grubas7 commented on SSDynamicText/NSAttributedString+SSTextSize.m:
```diff
@@ -0,0 +1,31 @@
+                                  if (value) {
+                                      UIFont *newFont = [UIFont fontWithDescriptor:value.fontDescriptor size:value.pointSize + delta];
+                                      [attributedString removeAttribute:NSFontAttributeName range:range];
+                                      [attributedString addAttribute:NSFontAttributeName value:newFont range:range];
```
You could provide static variable for `NSFontAttributeName`.

Remki replied:
It has no sense, NSFontAttributeName is already extern variable.

Grubas7 commented on SSDynamicText/SSDynamicButton.m:
```diff
@@ -20,6 +23,23 @@ - (void)setup;
+- (void)setAttributedTitle:(NSAttributedString *)title forState:(UIControlState)state {
+    NSNumber *key = @(state);
+    if (title) {
+        [self.baseAttributedTitlesDictionary setObject:title forKey:key];
```
You could use modern Obj-C:
`self.baseAttributedTitlesDictionary[key] = titile;`

Remki replied:
Well, in else statement i have to use `-removeObjectForKey` and for me this code looks better with `setObject:forKey`

ochococo replied:
Symmetry :+1:

Grubas7 commented on SSDynamicText/SSDynamicButton.m:
```diff
@@ -63,15 +83,31 @@ - (void)dealloc {
+}
+
+- (void)changeAttributedTitle:(NSAttributedString *)attributedTitle forState:(UIControlState)state withFontSizeDelta:(NSInteger)newDelta {
+    [super setAttributedTitle:[attributedTitle ss_attributedStringWithAdjustedFontSizeWithDelta:newDelta] forState:state];
```
Why `super`?

Grubas7 replied:
Generally a lot of code repetition, but I don't know how we could reduce this.

Remki replied:
Well, each class inherits from UIView, and UIView extension is a bad idea to reduce this code repetition. I think it should be left as is.

As for super, in [self setAttributedTitle] we change baseAttributedText, under which is based newDelta count. If we would call it when size changes - we would change baseAttributedText so delta would be always 0. I call here super to set new AttributedText without changing baseAttributedText.

Remki commented:
Ok. I have added some unit tests. I removed pod `Expecta`, since it wasn't used anywhere in the project. I added OCMock to change `UIApplication preferredContentSizeCategory` in tests. 
It's important to start unit tests with clean build, where content size is `UIContentSizeCategoryLarge`.

jhersh commented on Example/SSDynamicTextTests/SSDynamicTextTests.m:
```diff
@@ -34,20 +32,21 @@ - (void)tearDown {
+- (void)testApplicationFontSizeDelta {
+    //Assert
+    // This will probably fail if you've set a custom preferred font size in the simulator, please change it to UIContentSizeCategoryExtraLarge
     XCTAssertEqual(0, [[UIApplication sharedApplication] preferredFontSizeDelta], @"Font size delta should be zero");
```
Isn't `Large` the default, not `ExtraLarge`?

Grubas7 commented on Example/SSDynamicTextExample.xcodeproj/project.pbxproj:
```diff
@@ -85,6 +101,12 @@
+				B74328F81B946EEB0098E793 /* SSTextViewTests.m */,
+				B74328FA1B946FE90098E793 /* SSDynamicButtonTests.m */,
+				B74328FC1B9476010098E793 /* SSAttributedStringValidator.h */,
+				B74328FD1B9476010098E793 /* SSAttributedStringValidator.m */,
```
`SSAttributedStringValidator.{m,h}` should be in group `SSDynamicTextTests`?

Grubas7 commented on Example/SSDynamicTextExample/SSViewController.m:
```diff
@@ -9,68 +9,140 @@
+    } else {
+        [self setUpAttributedTexts];
+    }
+    self.isAttributedTextEnabled = !self.isAttributedTextEnabled;
```
Maybe move this line before `if ()`, because right now it looks:
attributedTextIsEnabled so set up normal text...

Grubas7 commented on Example/SSDynamicTextTests/SSDynamicButtonTests.m:
```diff
@@ -0,0 +1,85 @@
+
+static NSString * const kTestFontName = @"Avenir-Roman";
+static CGFloat    const kTestFontSize = 14.f;
+static CGFloat    const kTestFontSizeDifferenceForSizeEELarge = 2.f;
```
EELarge == XXL(arge)?
__Up to you__.

Grubas7 commented on Example/SSDynamicTextTests/SSTextViewTests.m:
```diff
@@ -0,0 +1,106 @@
+    [super tearDown];
+}
+
+- (void)mockEELargeCategory {
```
Every single test is using:
 `-mockEELargeCategory`
and 
`[[NSNotificationCenter defaultCenter] postNotificationName:UIContentSizeCategoryDidChangeNotification object:nil];`
So you could extract this to some helper, but __up to you__.
